src/Umbraco.Web/UmbracoComponentRenderer.cs

```diff
@@ -117,51 +117,7 @@ private IHtmlString RenderMacro(string alias, IDictionary<string, object> parame
                 x => x.Key.ToLowerInvariant(),
                 i => (i.Value is string) ? HttpUtility.HtmlDecode(i.Value.ToString()) : i.Value);
             
-            var macroControl = _macroRenderer.Render(alias, content, macroProps).GetAsControl();
-
-            string html;
-            if (macroControl is LiteralControl control)
```

Comment:
haha yes! all of this is totally gargabe in v8 🎉  nice find to get rid of legacy. `MacroContent.Control` is actually never used and is only ever assigned if GetAsControl() is used which then just creates an arbitrary LiteralControl. This will save a little bit of overhead for folks that use a ton of macros :)